src/Uno.SourceGenerationHost.Shared/ProjectLoader.cs

```diff
@@ -105,6 +105,14 @@ public static ProjectDetails LoadProjectDetails(BuildEnvironment environment)
 				properties["TargetFrameworkRootPath"] = environment.TargetFrameworkRootPath;
 			}
 
+			if (Environment.GetEnvironmentVariable("Platform") is string envPlatform && !string.IsNullOrEmpty(envPlatform))
+			{
+				throw new InvalidOperationException(
```

Comment:
An exception is pretty intense for such problem... are we sure it's breaking everything all the time?